internal/provider/recommendations_datasource_test.go

```diff
@@ -0,0 +1,123 @@
+package provider
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+)
+
+func TestAccRecommendationDatasource(t *testing.T) {
+	CheckAccTestsEnabled(t)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Read non-verbose.
+			{
+				Config: providerConfig + `
+data "grafana-adaptive-metrics_recommendations" "test" {}
+`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					checkMetricRecommendationAttr("data.grafana-adaptive-metrics_recommendations.test", "am_terraform_provider_acceptance_test_metric", "metric", "am_terraform_provider_acceptance_test_metric"),
```

Comment:
The metric name and the resource name is always the same. Could write a small helper in this test function to avoid those long lines
```
testAttr := func(attr, value string) resource.TestCheckFunc { return checkMetricRecommendationAttr("data.grafana-adaptive-metrics_recommendations.test", "am_terraform_provider_acceptance_test_metric", attr, value) }
```